lib/rubygems/specification.rb

```diff
@@ -1393,7 +1393,11 @@ def activate_dependencies
         raise e
       end
 
-      specs = spec_dep.to_specs
+      begin
+        specs = spec_dep.to_specs
+      rescue Gem::MissingSpecError => e
+        raise Gem::MissingSpecError.new(e.name, e.requirement, "at: #{self.spec_file}")
```

Comment:
Would it be possible the `raise` with the proper message from the beginning?

Reply:
https://github.com/rubygems/rubygems/blob/492496af849f564e96374671f3f3261452268e3d/lib/rubygems/dependency.rb#L311
Unfortunately we don't have access to the `runtime spec` trying to resolve it's dependencies inside `depency.rb` where the original error is raised.

Reply:
I see. Then I would raise a separate error like that represents that the runtime dependency `spec_dep` inside the `spec_file` gemspec couldn't be satisfied by any installed specifications, which I belieive is what's happening here, right?

Rescuing an exception and reraising the same exception, but with different arguments feels wrong to me.

Reply:
Also, does this same logic would apply to other `to_specs` calls in this file? What about `to_spec` calls?, which internally call `to_specs`?

I think this PR can work to fix the reported problem, but if we merge it as is, we could open a ticket to make a deeper refactoring of this situation.

Reply:
Definitively this can be improved! maybe we can optionally pass more context when creating a dependency so we can use spec information when needed in a dependency? 🤔